src/class.cc

```diff
@@ -12,14 +12,65 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include "class.h"
+
 namespace libwebp {
 
-// Let's implement stuff !
-// TODO: to create an animation:
-//       - create a WebPAnimEncoderOptions  config
-//       - set the quality to something in your search space
-//       - call WebPAnimEncoderNew, then fill the animation with WebPPicture
-//         using WebPAnimEncoderAdd
-//       - finalize with WebPAnimEncoderAssemble
+Thumbnailer::Thumbnailer() {
+  WebPAnimEncoderOptionsInit(&anim_config);
+  WebPConfigInit(&config);
+}
+
+Thumbnailer::~Thumbnailer() { WebPAnimEncoderDelete(enc); }
+
+bool Thumbnailer::AddFrame(const WebPPicture& pic, int timestamp_ms) {
+  bool ok = true;
+
+  // Verifies dimension of frames.
+  if (frames.empty()) {
+    enc = WebPAnimEncoderNew(pic.width, pic.height, &anim_config);
+    ok = (enc != nullptr);
+  } else {
+    if (pic.width != (frames[0].pic).width ||
+        pic.height != (frames[0].pic).height) {
+      ok = false;
+    }
+  }
+
```

Comment:
```suggestion
if (!ok) return false;
```
(that is: don't add the frame if something went wrong before)